Src/ILGPU/IR/Analyses/PointerAlignments.cs

```diff
@@ -150,6 +150,66 @@ public static AlignmentInfo Apply(Method rootMethod, int globalAlignment)
             return new AlignmentInfo(result);
         }
 
+        /// <summary>
+        /// Tries to determine power of 2 information for the given unary operation.
+        /// </summary>
+        /// <param name="unary">The unary operation to analyze.</param>
+        /// <returns>The power of 2 value (if any).</returns>
+        private static int? TryGetPowerOf2(UnaryArithmeticValue unary) =>
+            unary.Kind switch
+            {
+                UnaryArithmeticKind.Abs => TryGetPowerOf2(unary.Value),
+                _ => null
+            };
+
+        /// <summary>
+        /// Tries to determine power of 2 information for the given binary operation.
+        /// </summary>
+        /// <param name="binary">The binary operation to analyze.</param>
+        /// <returns>The power of 2 value (if any).</returns>
+        private static int? TryGetPowerOf2(BinaryArithmeticValue binary) =>
+            binary.Kind switch
+            {
+                // Check whether either the left or the right operand are a power of 2
+                BinaryArithmeticKind.Mul =>
+                    TryGetPowerOf2(binary.Left) ??
+                    TryGetPowerOf2(binary.Right),
+                // Check whether we can determine a power of 2 of the left operand or
+                // whether the RHS of the SHL operation is a primitive value
+                BinaryArithmeticKind.Shl =>
+                    TryGetPowerOf2(binary.Left) ??
+                    (binary.Right.Resolve() is PrimitiveValue shlPrimitive &&
+                    shlPrimitive.Int32Value > 0
+                    ? (int?)(shlPrimitive.Int32Value * 2)
+                    : null),
+                _ => null,
+            };
+
+        /// <summary>
+        /// Tries to determine a power of 2 for the given value (if any could be
+        /// determined).
+        /// </summary>
+        /// <param name="value">The value to analyze.</param>
+        /// <returns>The power of 2 value (if any).</returns>
+        private static int? TryGetPowerOf2(Value value)
+        {
+            // Ensure that the value is operating on an integer type
+            value.Assert(value.BasicValueType.IsInt());
+            return value switch
+            {
+                // Check whether the value is a power of two in the case of a raw value
+                PrimitiveValue primitive =>
```

Comment:
We "automagically" handle long types as well since the 32-bit part of a long will be taken into account. However, this implicitly limits the maximum possible alignment to 2^32-1 (which should not hurt in practice 😄)